### PR TITLE
Drop Travis in favor of Github workflow

### PR DESCRIPTION
**Pull request by sfdye**
- [x] Drop Travis
- [x] Update `DEPLOY.md`

### CODE REVIEW

**Comment by sfdye** [comment]
I will hand over this branch to you @s-t-e-v-e-n-k , after you have the codecov changes

File: DEPLOY.md [hunk 1]
```diff
@@ -9,10 +9,11 @@ Travis-ci will push tagged commits to PyPI. Here are the steps:
 Next version number? (previous: 'XXX')
 ```
 
-2. Give the new version number based on previous version (see semantic release)
+2. Give the new version number based on previous version (Use semantic versioning)
 
-3. Now the push will be on hold until you press Enter. Manually inpect the changelog (`doc/changes.rst`) to make changes if necessary. Once you are sure, go back and press Enter. 
+3. Create a new Github [release](https://github.com/PyGithub/PyGithub/releases) from the tag that'sjust committed, with the same release note from `doc/changes.rst`. This step is the hook that will trigger the workflow. (also needed for some web spiders for changelog parsing)
```
Comment (s-t-e-v-e-n-k) [inline DEPLOY.md:14]: "that'sjust" => "that has just been committed" to correct the missing space and I think it scans better.

Comment (sfdye) [reply DEPLOY.md:14]: You can use the suggest feature

**Comment by sfdye** [comment]
One thing though, do we really want to run codecov for all python version?

**Comment by s-t-e-v-e-n-k** [comment]
I did wonder that, shall I ratchet it down to py3.6 only?

**Comment by sfdye** [comment]
TIL we can actually combine them:
https://coverage.readthedocs.io/en/coverage-5.0.4/cmd.html#combining-data-files

and upload one file together

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye I've taken the easy way out and we skip it for every version except 3.6, and it's one line! WDYT?

**Comment by sfdye** [comment]
Merge it when you feel like

**Pull request merged**